src/rules/tsUtils.ts

```diff
@@ -320,6 +320,185 @@ export const isExpectCallWithParent = (
+): ParsedExpectMatcher => ({
+  ...parsedMember,
+  /**
+   * The arguments being passed to this `Matcher`, if any.
+   *
+   * If this matcher isn't called, this will be `null`.
+   */
+  arguments:
+    /* istanbul ignore next */
+    parsedMember.node.parent &&
+    parsedMember.node.parent.type === AST_NODE_TYPES.CallExpression
+      ? parsedMember.node.parent.arguments
+      : null,
+  // isNamed(...names: string[]) {
+  //   return names.includes(this.name);
+  // },
+});
+
+/**
+ * Re-parses the given `parsedMember` as a `ParsedExpectModifier`.
+ *
+ * If the given `parsedMember` does not have a `name` of a valid `Modifier`,
+ * an exception will be thrown.
+ *
+ * @param {ParsedExpectMember<ModifierName>} parsedMember
+ *
+ * @return {ParsedExpectModifier}
+ */
+const reparseMemberAsModifier = (
+  parsedMember: ParsedExpectMember<ModifierName>,
+): ParsedExpectModifier => {
+  if (isSpecificMember(parsedMember, ModifierName.not)) {
+    return parsedMember;
+  }
+
+  /* istanbul ignore if */
+  if (
+    !isSpecificMember(parsedMember, ModifierName.resolves) &&
+    !isSpecificMember(parsedMember, ModifierName.rejects)
+  ) {
+    throw new Error(
+      `modifier name must be either "${ModifierName.resolves}" or "${ModifierName.rejects}" (got "${parsedMember.name}")`,
+    ); // ts doesn't think that the ModifierName.not check is the direct inverse as the above two checks
+  } // todo: impossible at runtime, but can't be typed w/o negation support
+
+  /* istanbul ignore next */
+  const negation =
+    parsedMember.node.parent &&
+    isExpectMember(parsedMember.node.parent, ModifierName.not)
+      ? parsedMember.node.parent
+      : undefined;
+
+  return {
+    ...parsedMember,
+    negation,
+  };
+};
+
+const isSpecificMember = <Name extends ExpectPropertyName>(
+  member: ParsedExpectMember,
+  specific: Name,
+): member is ParsedExpectMember<Name> => member.name === specific;
+
+/**
+ * Checks if the given `ParsedExpectMember` should be re-parsed as an `ParsedExpectModifier`.
+ *
+ * @param {ParsedExpectMember} member
+ *
+ * @return {member is ParsedExpectMember<ModifierName>}
+ */
+const shouldBeParsedExpectModifier = (
+  member: ParsedExpectMember,
+): member is ParsedExpectMember<ModifierName> =>
+  ModifierName.hasOwnProperty(member.name);
+
+export const parseExpectCall = <ExpectNode extends ExpectCall>(
+  expect: ExpectNode,
+): Expectation<ExpectNode> => {
+  const expectation: Expectation<ExpectNode> = {
+    expect,
+  };
+
+  if (!isExpectMember(expect.parent)) {
+    return expectation;
+  }
+
+  const parsedMember = parseExpectMember(expect.parent);
+  if (!shouldBeParsedExpectModifier(parsedMember)) {
+    expectation.matcher = reparseAsMatcher(parsedMember);
+
+    return expectation;
+  }
+
+  const modifier = (expectation.modifier = reparseMemberAsModifier(
+    parsedMember,
+  ));
+
+  const memberNode =
+    ('negation' in modifier && modifier.negation) || modifier.node;
```

Comment:
```suggestion
    modifier.negation || modifier.node;
```

Reply:
Good catch - that's not required anymore as it's become type safe :D